include/zephyr/drivers/cellular.h

```diff
@@ -52,6 +52,37 @@ struct cellular_network {
 	uint16_t size;
 };
 
+enum cellular_signal_type {
+	CELLULAR_SIGNAL_RSSI,
+	CELLULAR_SIGNAL_QUALITY,
+	CELLULAR_SIGNAL_PERCENTAGE,
```

Comment:
I like this approach, I think we should use the technical values here, RSSI, RSRP and RSRQ, which seem to cover both GSM and LTE very well, and which are quite mathematically appropriate :)